src/storage/txn/tablet_flow_controller.rs

```diff
@@ -0,0 +1,727 @@
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        assert_eq!(flow_controller.consume(region_id, 2000), Duration::ZERO);
+        loop {
+            if flow_controller.total_bytes_consumed(region_id) == 0 {
+                break;
+            }
+            std::thread::sleep(TICK_DURATION);
+        }
+
+        // exceeds the threshold on start
+        stub.0.num_memtable_files.store(8, Ordering::Relaxed);
+        tx.send(FlowInfo::Flush(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        tx.send(FlowInfo::L0Intra(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        assert_eq!(flow_controller.should_drop(region_id), false);
+        // on start check forbids flow control
+        assert_eq!(flow_controller.is_unlimited(region_id), true);
+        // once falls below the threshold, pass the on start check
+        stub.0.num_memtable_files.store(1, Ordering::Relaxed);
+        tx.send(FlowInfo::Flush(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        tx.send(FlowInfo::L0Intra(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        // not throttle when the average of the sliding window doesn't exceeds the threshold
+        stub.0.num_memtable_files.store(6, Ordering::Relaxed);
+        tx.send(FlowInfo::Flush(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        tx.send(FlowInfo::L0Intra(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        assert_eq!(flow_controller.should_drop(region_id), false);
+        assert_eq!(flow_controller.is_unlimited(region_id), true);
+
+        // the average of sliding window exceeds the threshold
+        stub.0.num_memtable_files.store(6, Ordering::Relaxed);
+        tx.send(FlowInfo::Flush(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        tx.send(FlowInfo::L0Intra(
+            "default".to_string(),
+            0,
+            region_id,
+            tablet_suffix,
+        ))
+        .unwrap();
+        assert_eq!(flow_controller.should_drop(region_id), false);
+        assert_eq!(flow_controller.is_unlimited(region_id), false);
+        assert_ne!(flow_controller.consume(region_id, 2000), Duration::ZERO);
+
+        // not throttle once the number of memtables falls below the threshold
+        stub.0.num_memtable_files.store(1, Ordering::Relaxed);
```

Comment:
num_memtables?

Reply:
That's the existing code, but i agree with you.